src/System.Runtime/tests/System/GCTests.cs

```diff
@@ -157,14 +169,19 @@ public static void KeepAlive_Null()
 
         private class KeepAliveNullTest
         {
-            public static void Run()
+            [System.Runtime.CompilerServices.MethodImplAttribute(System.Runtime.CompilerServices.MethodImplOptions.NoInlining)]
+            private static void MakeAndNull()
             {
                 var obj = new TestObject();
                 obj = null;
+            }
+
+            public static void Run()
+            {
+                MakeAndNull();
 
                 GC.Collect();
                 GC.WaitForPendingFinalizers();
-                GC.KeepAlive(obj);
 
```

Comment:
I confess I actually don't understand some of this stuff completely - for example, in one test it set a local variable to null and right below it called KeepAlive or something similar on it; no idea what the purpose was. If you think that this should be changed in a different way that's totally cool but it would be easier if you could be more specific, please. Oh and as far as how do these tests work - before or now - someone else would have to answer that; some of this stuff is way beyond my comprehension, I am afraid ;).